executor/builder.go

```diff
@@ -235,6 +235,21 @@ func (b *executorBuilder) buildIndexScan(v *plan.IndexScan) Executor {
 		return nil
 	}
 	tbl, _ := b.is.TableByID(v.Table.ID)
+
+	var idx table.Index
+	for _, val := range tbl.Indices() {
+		if val.Meta().Name.L == v.Index.Name.L {
+			idx = val
+			break
+		}
+	}
+
+	valueTypes := make([]*types.FieldType, len(idx.Meta().Columns))
```

Comment:
We can just use `tbl.Cols()` in `refineRange`, we don't need to create a slice of `*types.FieldType`.